docs/admin/templates/extending-templates/parameters.md

```diff
@@ -391,3 +391,546 @@ parameters in one of two ways:
+
+For templates serving multiple teams or use cases, you can create comprehensive branching paths:
+
+```tf
+data "coder_parameter" "environment_type" {
+  name         = "environment_type"
+  display_name = "Environment Type"
+  description  = "Select your preferred development environment"
+  type         = "string"
+  default      = "container"
+
+  option {
+    name  = "Container"
+    value = "container"
+  }
+
+  option {
+    name  = "Virtual Machine"
+    value = "vm"
+  }
+}
+
+# Container-specific parameters
+data "coder_parameter" "container_image" {
+  name         = "container_image"
+  display_name = "Container Image"
+  description  = "Select a container image for your environment"
+  type         = "string"
+  default      = "ubuntu:latest"
+
+  # Only show when container environment is selected
+  condition {
+    field = data.coder_parameter.environment_type.name
+    value = "container"
+  }
+
+  option {
+    name  = "Ubuntu"
+    value = "ubuntu:latest"
+  }
+
+  option {
+    name  = "Python"
+    value = "python:3.9"
+  }
+}
+
+# VM-specific parameters
+data "coder_parameter" "vm_image" {
+  name         = "vm_image"
+  display_name = "VM Image"
+  description  = "Select a VM image for your environment"
+  type         = "string"
+  default      = "ubuntu-20.04"
+
+  # Only show when VM environment is selected
+  condition {
+    field = data.coder_parameter.environment_type.name
+    value = "vm"
+  }
+
+  option {
+    name  = "Ubuntu 20.04"
+    value = "ubuntu-20.04"
+  }
+
+  option {
+    name  = "Debian 11"
+    value = "debian-11"
+  }
+}
+```
+
+</details>
+
+<details><summary>Conditional Validation</summary>
+
+Adjust validation rules dynamically based on parameter values:
+
+```tf
+data "coder_parameter" "team" {
+  name        = "team"
+  display_name = "Team"
+  type        = "string"
+  default     = "engineering"
+
+  option {
+    name  = "Engineering"
+    value = "engineering"
+  }
+
+  option {
+    name  = "Data Science"
+    value = "data-science"
+  }
+}
+
+data "coder_parameter" "cpu_count" {
+  name        = "cpu_count"
+  display_name = "CPU Count"
+  type        = "number"
+  default     = 2
+
+  # Engineering team has lower limits
+  dynamic "validation" {
+    for_each = data.coder_parameter.team.value == "engineering" ? [1] : []
+    content {
+      min = 1
+      max = 4
+    }
+  }
+
+  # Data Science team has higher limits
+  dynamic "validation" {
+    for_each = data.coder_parameter.team.value == "data-science" ? [1] : []
+    content {
+      min = 2
+      max = 8
+    }
+  }
+}
+```
+
+</details>
+
+## API Details
+
+Dynamic Parameters uses WebSocket communication to provide real-time updates.
+
+The WebSocket endpoint is:
+
+```api
+GET /users/{user}/templateversions/{templateversion}/parameters
+```
+
+The WebSocket connection sends parameter changes as they occur and receives updated parameter state and validation errors in response.
```

Comment:
Let's not add this right now until GA. The API should be considered mutable until then

```suggestion
```